ShellPkg/Application/Shell/ShellProtocol.c

```diff
@@ -735,52 +742,58 @@ EfiShellGetDeviceName (
     //
     // Now check the parent controller using this as the child.
     //
-    Status = PARSE_HANDLE_DATABASE_PARENTS (DeviceHandle, &ParentControllerCount, &ParentControllerBuffer);
-    if ((DeviceNameToReturn == NULL) && !EFI_ERROR (Status)) {
+    if (DeviceNameToReturn == NULL) {
+      Status = PARSE_HANDLE_DATABASE_PARENTS (DeviceHandle, &ParentControllerCount, &ParentControllerBuffer);
+      if (EFI_ERROR (Status)) {
+        ParentControllerCount = 0;
+      }
+
       for (LoopVar = 0; LoopVar < ParentControllerCount; LoopVar++) {
-        Status = PARSE_HANDLE_DATABASE_UEFI_DRIVERS (ParentControllerBuffer[LoopVar], &ParentDriverCount, &ParentDriverBuffer);
-        if (!EFI_ERROR (Status)) {
-          for (HandleCount = 0; HandleCount < ParentDriverCount; HandleCount++) {
-            //
-            // try using that driver's component name with controller and our driver as the child.
-            //
+        PARSE_HANDLE_DATABASE_UEFI_DRIVERS (ParentControllerBuffer[LoopVar], &ParentDriverCount, &ParentDriverBuffer);

```

Comment:
Shouldn't we check the returned Status as in the original code ?

Reply:
Yes, I think this was a bad merge on our part in the past. Will restore the status checking, thanks for catching.